syntax = "proto3";

service MessagingService {
  rpc RegisterPublicKey (PublicKey) returns (Response);
  rpc RequestPublicKey (ClientRequest) returns (PublicKey);
  rpc SendMessage (Message) returns (Response);
  rpc ReceiveMessages (ClientRequest) returns (stream Message);
}

message PublicKey {
  string sender = 1;  // ID del cliente que envía la llave pública
  uint64 e = 2;        // Exponente público
  uint64 n = 3;        // Módulo público
}

message ClientRequest {
  string sender = 1;  // ID del cliente solicitante
}

message Message {
  string sender = 1;    // ID del cliente que envía el mensaje
  string receiver = 2;  // ID del cliente que debe recibir el mensaje
  bytes content = 3;    // Contenido cifrado
}

message Response {
  string status = 1;
}